python/ray/train/tests/test_gpu.py

```diff
@@ -69,6 +91,40 @@ def train_fn():
         )
 
 
+# TODO: Refactor as a backend test.
+@pytest.mark.parametrize("num_gpus_per_worker", [0.5, 1, 2])
+def test_torch_get_device_dist(ray_2_node_4_gpu, num_gpus_per_worker):
+    @patch("torch.cuda.is_available", lambda: True)
+    def train_fn():
+        return train.torch.get_device().index
+
+    trainer = Trainer(
+        TorchConfig(backend="gloo"),
+        num_workers=int(8 / num_gpus_per_worker),
+        use_gpu=True,
+        resources_per_worker={"GPU": num_gpus_per_worker},
+    )
+    trainer.start()
+    devices = trainer.run(train_fn)
+    trainer.shutdown()
+
+    count = Counter(devices)
+    if num_gpus_per_worker == 0.5:
+        for i in range(4):
+            assert count[i] == 4
+    elif num_gpus_per_worker == 1:
+        for i in range(4):
+            assert count[i] == 2
+    elif num_gpus_per_worker == 2:
+        for i in range(2):
+            assert count[2 * i] == 2
```

Comment:
**But the fake cluster is still running on a GPU instance, no?**

i guess no, because i am still able to run locally.


**For each of these cases, can you add a comment on what the expected topology should be? Mayb we can also scope this down to ray_2_node_2_gpu for simplicity.**

i see, sounds good to me.

Reply:
by the way, for example 
```python
# CUDA_VISIBLE_DEVICE = '0,1,2,3'
# num_gpus_per_worker == 2
```

the `ray_gpu_ids` will always first use `0,1` and then `2,3`? 

On the other hand, will it possible to for it use `0,2` and then `1,3`? (in this case, then my answer will be different. )